code/modules/admin/view_variables/debug_variable_appearance.dm

```diff
@@ -0,0 +1,236 @@
+		if(NAMEOF(appearance, maptext))
+			return appearance.maptext
+		if(NAMEOF(appearance, maptext_width))
+			return appearance.maptext_width
+		if(NAMEOF(appearance, maptext_height))
+			return appearance.maptext_height
+		if(NAMEOF(appearance, maptext_x))
+			return appearance.maptext_x
+		if(NAMEOF(appearance, maptext_y))
+			return appearance.maptext_y
+		if(NAMEOF(appearance, mouse_over_pointer))
+			return appearance.mouse_over_pointer
+		if(NAMEOF(appearance, mouse_drag_pointer))
+			return appearance.mouse_drag_pointer
+		if(NAMEOF(appearance, mouse_drop_pointer))
+			return appearance.mouse_drop_pointer
+		if(NAMEOF(appearance, mouse_drop_zone))
+			return appearance:mouse_drop_zone
+		if(NAMEOF(appearance, mouse_opacity))
+			return appearance.mouse_opacity
+		if(NAMEOF(appearance, name))
+			return appearance.name
+		if(NAMEOF(appearance, opacity))
+			return appearance.opacity
+		if(NAMEOF(appearance, overlays))
+			return appearance.overlays
+		if("override") // only /image has this. mocking type can't steal byond internal var name
+			var/image/image_appearance = appearance
+			return image_appearance.override
+		if(NAMEOF(appearance, pixel_x))
+			return appearance.pixel_x
+		if(NAMEOF(appearance, pixel_y))
+			return appearance.pixel_y
+		if(NAMEOF(appearance, pixel_w))
+			return appearance.pixel_w
+		if(NAMEOF(appearance, pixel_z))
+			return appearance.pixel_z
+		if(NAMEOF(appearance, plane))
+			return appearance.plane
+		if(NAMEOF(appearance, render_source))
+			return appearance.render_source
+		if(NAMEOF(appearance, render_target))
+			return appearance.render_target
+		if(NAMEOF(appearance, suffix))
+			return appearance.suffix
+		if(NAMEOF(appearance, text))
+			return appearance.text
+		if(NAMEOF(appearance, transform))
+			return appearance.transform
+		if(NAMEOF(appearance, underlays))
+			return appearance.underlays
+
+		if(NAMEOF(appearance, parent_type))
+			return appearance.parent_type
+		if(NAMEOF(appearance, type))
+			return /image/appearance // don't fool people
+
+		// These are not documented ones but trackable values. Maybe we'd want these.
+		if(NAMEOF(appearance, animate_movement))
+			return appearance.animate_movement
+		if(NAMEOF(appearance, dir))
+			return appearance.dir
+		if(NAMEOF(appearance, glide_size))
+			return appearance.glide_size
+		if("pixel_step_size")
+			return "" //atom_appearance.pixel_step_size
+			// DM compiler complains this
+
+		// I am not sure if these will be ever detected, but I made a connection just in case.
+		if(NAMEOF(appearance, contents)) // It's not a thing, but I don't believe how DM will change /appearance in future.
+			return appearance.contents
+		if(NAMEOF(appearance, loc)) // same reason above
+			return appearance.loc
+		if(NAMEOF(appearance, vis_contents)) // same reason above
+			return appearance.vis_contents
+		if(NAMEOF(appearance, vis_flags)) // DM document says /appearance has this, but it throws error
+			return appearance.vis_flags
```

Comment:
for the record, the way this actually works is very dumb
appearances carry vis_flags, but you can't read them directly off
what you CAN do is copy the appearance onto a MOVABLE, and read the vis_flags off THAT. v stupid am sorry

Reply:
```
        var/static/atom/movable/pluto = new() // all alone at the end of the universe
        pluto.appearance = appearance_father // We copy over our appearance onto an atom. This is done so we can read vars carried by but not accessible on appearances
        _vis_flags = pluto.vis_flags
        appearance_ref = REF(appearance_father)
```
like this ya feel? (I am assuming you're keeping something of this form downstream somewhere, this'll make that less hellish)